src/witan/app/forecast.clj

```diff
@@ -103,9 +105,33 @@
 
 ;;;;;;
 
+;;;;;; NOTES:
+;; - We currently don't use `exists?` properly, but we should in the context of
+;;   re-naming forecasts. Currently, duplicate owner+name combinations return
+;;   304.
+
 (defresource forecasts
+  util/json-resource
   :allowed-methods #{:get :post}
-  :available-media-types ["application/json"]
-  :handle-ok (fn [_] (s/validate
-                      [ws/Forecast]
-                      (map ->Forecast (get-forecasts)))))
+  :processable? (util/post!-processable-validation ws/NewForecast)
+  :exists? (fn [ctx]
+             (if (= :post (-> ctx :request :request-method))
```

Comment:
I bet `(= :post (-> ctx :request :request-method)` will want to be a little helper function somewhere `http-post?`